src/lib/Search/Legacy/Content/Common/Gateway/CriterionHandler/FieldValue/Handler.php

```
@@ -72,12 +73,16 @@ public function handle(
         Criterion $criterion,
         string $column
     ) {
+        if (is_array($criterion->value) && !in_array($criterion->operator, [Criterion\Operator::IN, Criterion\Operator::BETWEEN])) {
+            $criterion->value = current($criterion->value);
+        }
```

Passing an array value for Criterion with mismatched operator is an error and we should not attempt to recover from it.
If anything we should throw an exception.